answerking_app/utils/mixins/GenericMixins.py

```diff
@@ -1,32 +1,34 @@
-from MySQLdb.constants.ER import DUP_ENTRY
+from typing import NoReturn
+
 from django.db import IntegrityError
-from rest_framework import status
+from django.http import Http404
+from MySQLdb.constants.ER import DUP_ENTRY
 from rest_framework.mixins import CreateModelMixin, UpdateModelMixin
 from rest_framework.request import Request
 from rest_framework.response import Response
 
+from answerking_app.utils.mixins.ApiExceptions import (Http400BadRequest,
+                                                       ServiceUnavailable)
+
 
 class CreateMixin(CreateModelMixin):
     def create(self, request: Request, *args, **kwargs) -> Response:
         try:
             return super().create(request, *args, **kwargs)
         except IntegrityError as exc:
-            return duplicate_check(exc)
+            handle_IntegrityError(exc)
 
 
 class UpdateMixin(UpdateModelMixin):
     def update(self, request: Request, *args, **kwargs) -> Response:
         try:
             return super().update(request, *args, **kwargs)
         except IntegrityError as exc:
-            return duplicate_check(exc)
+            handle_IntegrityError(exc)
 
 
-def duplicate_check(exc: IntegrityError) -> Response:
+def handle_IntegrityError(exc: IntegrityError) -> NoReturn:
```

Comment:
would `duplicate_check `or `dup_entry_check `not be a better name for this as we are checking for a specific type of integrity error (duplicate entry) rather than a generic Integrity error?